test/src/test/java/org/corfudb/runtime/collections/CorfuStoreShimTest.java

```diff
@@ -1418,6 +1418,7 @@ public void testRegistryTableUpdateOnRecordChange() throws Exception {
         assertThat(options.getStreamTagCount()).isEqualTo(2);
         assertThat(options.getStreamTag(0)).isEqualTo("sample_streamer_2");
         assertThat(options.getStreamTag(1)).isEqualTo("sample_streamer_3");
+        assertThat(options.getLogicalGroup()).isEqualTo("group1");
```

Comment:
Should we include the client name as well?
Also I am wondering if we should have a flag as well (like isFederated) for logical groups use case.
My thought is that the design would then be uniform: a flag to indicate which replication model to use and then followed by the model specific options.
The flags can be an enum. 
It can make the dynamicStreams part, find streams for a model, also consistent.